jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -256,13 +257,168 @@ public synchronized void closeTracer() {
    */
   @Deprecated
   public void setStatsFactor(StatsFactory statsFactory) {
-    this.statsFactory = statsFactory;
+    setStatsFactory(statsFactory);
   }
 
   public void setStatsFactory(StatsFactory statsFactory) {
     this.statsFactory = statsFactory;
   }
 
+  public String getServiceName() {
+    return serviceName;
+  }
+
+  public SamplerConfiguration getSamplerConfig() {
+    return samplerConfig;
+  }
+
+  public ReporterConfiguration getReporterConfig() {
+    return reporterConfig;
+  }
+
+  public CodecConfiguration getCodecConfig() {
+    return codecConfig;
+  }
+
+  public StatsFactory getStatsFactory() {
+    return statsFactory;
+  }
+
+  /**
+   * Starts a new builder with the given service name.
+   *
+   * @param serviceName the service name
+   * @return a properly started builder
+   */
+  public static Builder builder(String serviceName) {
+    return new Builder(serviceName);
+  }
+
+  /**
+   * Starts a new builder. Note that a service name is required to obtain a Tracer. If the service name is provided
+   * via an environment variable, make sure to call {@link Builder#fromEnv()}.
+   *
+   * @return a builder that still requires a name to start
+   */
+  public static Builder builder() {
+    return new Builder(null);
```

Comment:
This looks a bit inconsistent. Builder constructor requires service name and this not.

Does it make sense to change `fromEnv()` to be static?

Reply:
You mean, `Builder.fromEnv()` to be the equivalent of `Builder.builder()`, but requiring the service name to come from the env var? Sounds like a good solution to me.

Reply:
What about `Configuration.builderFromEnv().build()`?

Not sure I like the idea of `fromEnv()` overwriting any config previously setup - seems pointless. This would also mean that we only have the `builder(serviceName)` option.

Reply:
> You mean, Builder.fromEnv() to be the equivalent of Builder.builder(), but requiring the service name to come from the env var?

yes `Configuration.Builder = Configuration.Builder.fromEnv() ` (or just configuration if configuration is a builder)

Reply:
Even if it's inconsistent with the `Configuration.builder()` method that starts a builder?

```
Configuration configuration = Configuration.builder().with...().build();
```

vs.

```
Configuration configuration = Configuration.Builder.fromEnv().build();
```

Between this and @objectiser, I think his is more consistent.

Reply:
> Even if it's inconsistent with the Configuration.builder() method that starts a builder?

I am not sure if I understand you. I started this thread because I don't like that `Configuration.builder()` creates a builder with `null` service name. `Builder.fromEnv()` without param is fine because it expects service name to be specified in env. Or we can remove service name from builder constructor and use `unknown` (Zipkin does this).

Reply:
As part of changing the `Configuration.builder(String)` to add `Configuration.builderFromEnv()`, I removed the `fromEnv()` from the builder.

About removing the service name requirement, making it default to `unknown`, I would rather have it as part of another PR, especially because it changes an existing behavior.